pkg/operator/ceph/cluster/crash/add.go

```diff
@@ -0,0 +1,164 @@
+	if err != nil {
+		return fmt.Errorf("failed to watch for node changes. %+v", err)
+	}
+
+	// Watch for changes to the ceph-crash deployments
+	logger.Debugf("watch for changes to the ceph-crash deployments")
+	err = c.Watch(
+		&source.Kind{Type: &appsv1.Deployment{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+				deployment, ok := obj.Object.(*appsv1.Deployment)
+				if !ok {
+					return []reconcile.Request{}
+				}
+				labels := deployment.GetLabels()
+				appName, ok := labels[k8sutil.AppAttr]
+				if !ok || appName != AppName {
+					return []reconcile.Request{}
+				}
+				nodeName, ok := deployment.Spec.Template.ObjectMeta.Labels[NodeNameLabel]
+				if !ok {
+					return []reconcile.Request{}
+				}
+				req := reconcile.Request{NamespacedName: types.NamespacedName{Name: nodeName}}
+				return []reconcile.Request{req}
+			}),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to watch for changes on the ceph-crash deployment. %+v", err)
+	}
+
+	// Watch for changes to the ceph pod nodename and enqueue their nodes
+	logger.Debugf("watch for changes to the ceph pod nodename and enqueue their nodes")
+	err = c.Watch(
+		&source.Kind{Type: &corev1.Pod{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(obj handler.MapObject) []reconcile.Request {
+				pod, ok := obj.Object.(*corev1.Pod)
+				if !ok {
+					return []reconcile.Request{}
+				}
+				nodeName := pod.Spec.NodeName
+				if nodeName == "" {
+					return []reconcile.Request{}
+				}
+				if isCephPod(pod.Labels, pod.Name) {
+					req := reconcile.Request{NamespacedName: types.NamespacedName{Name: nodeName}}
+					return []reconcile.Request{req}
+				}
+				logger.Debugf("%q is not a ceph pod", pod.Name)
+				return []reconcile.Request{}
+			}),
+		},
+		// only enqueue the update event if the pod moved nodes
+		predicate.Funcs{
+			UpdateFunc: func(event event.UpdateEvent) bool {
+				oldPod, ok := event.ObjectOld.(*corev1.Pod)
+				if !ok {
+					return false
+				}
+				newPod, ok := event.ObjectNew.(*corev1.Pod)
+				if !ok {
+					return false
+				}
+				// only enqueue if the nodename has changed
+				if oldPod.Spec.NodeName == newPod.Spec.NodeName {
+					return false
+				}
+				return true
+			},
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to watch for changes on the ceph pod nodename and enqueue their nodes. %+v", err)
+	}
+
+	return nil
+}
+
+func isCephPod(labels map[string]string, podName string) bool {
+	_, ok := labels["rook_cluster"]
+	// canary pods for monitors might stick around during startup
+	// at that time, the initial monitors haven't been deployed yet.
+	// If we don't invalidate canary pods,
+	// the crash collector pod will start and environment variable like 'ROOK_CEPH_MON_HOST'
+	// will be empty since the monitors don't exist yet
+	isCanaryPod := strings.Contains(podName, "-canary-")
```

Comment:
What's the resolution on using label/suffix for this? Tentatively approving as long as this hasn't simply slipped through

Reply:
@leseb did you look at using `strings.HasSuffix(podName, "-canary")`?

Reply:
I did look into it but the pod name is rook-ceph-mon-a-canary-545897654f-dvz28 so this doesn't quite work

this would work for the deployment name but not the pod name

Reply:
@rohantmp how strongly do you feel about this? I need to followup this PR with a client.crash key, can I do this too in that next PR? Thanks!